Neos.Neos/Classes/Domain/SoftRemoval/SoftRemovedTag.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Neos\Neos\Domain\SoftRemoval;
+
+use Neos\ContentRepository\Core\Feature\NodeRemoval\Command\RemoveNodeAggregate;
+use Neos\ContentRepository\Core\Feature\SubtreeTagging\Command\TagSubtree;
+use Neos\ContentRepository\Core\Feature\SubtreeTagging\Dto\SubtreeTag;
+use Neos\ContentRepository\Core\Projection\ContentGraph\VisibilityConstraints;
+use Neos\Neos\Domain\Service\WorkspacePublishingService;
+use Neos\Flow\Annotations as Flow;
+
+/**
+ * Content repository subtree tag which is used to denote that a node is soft removed
+ *
+ * Issuing 'hard' removals via {@see RemoveNodeAggregate} on a non-live workspace ist not desired in Neos
+ * and comes with complications:
+ *
+ * - Hard removals destroy all hierarchy information immediately, making it impossible to locate where a removal took place.
+ *   A special case imposes the deletion of newly created nodes.
+ *   Associating content changes with its document is important when publishing via {@see WorkspacePublishingService::publishChangesInDocument()}
+ *
+ * - Hard removals easily cause conflicts:
+ *   The removal of a selected node might not be published in that scope if a node was moved outwards.
+ *   The rebase will cause conflicts if changes were made in a hierarchy of removed nodes, like moving nodes outwards.
+ *
+ * Instead, tagging a node / its subtree via {@see TagSubtree} will mark it as removed in Neos.
+ *
+ *     $contentRepository->handle(TagSubtree::create(
+ *         $node->workspaceName,
+ *         $node->aggregateId,
+ *         $node->dimensionSpacePoint,
+ *         NodeVariantSelectionStrategy::STRATEGY_ALL_SPECIALIZATIONS,
+ *         SoftRemovedTag::getSubtreeTag()
+ *     ));
+ *
+ * Nodes tagged as removed will not show up in the frontend rendering nor in the backend by default.
+ * Only subtracting this removed tag from the {@see VisibilityConstraints} or operating on the node aggregates directly
+ * will make soft removed nodes available.
+ *
+ * @Flow\Proxy(false)
+ */
+final readonly class SoftRemovedTag
```

Comment:
I haven't understood why we do
```php
SubtreeTag::disabled()
```
but
```
SoftRemovedTag::getSubtreeTag()
```

Yet another abstraction is not really helpful IMO

What about simply doing sth like:

```php
final readonly class NeosSubtreeTags {
    public static function disabled(): SubtreeTag {
        return SubtreeTag::fromString('disabled');
    }

    public static function removed(): SubtreeTag {
        return SubtreeTag::fromString('removed');
    }
}
```
?

This way we can make this a pure Neos concept and still avoid the magic strings to spread.

Instead of `SoftRemovedTag::isRemovedSubtreeTag($tag)` you'd write `$tag->equals(NeosSubtreeTags::removed())`

Reply:
the suggestion to fully move out the `disabled` concern from the core is what i do really like. It also is what convinced me to implement: https://github.com/neos/neos-development-collection/pull/5496

I introduced a `NeosSubtreeTag` as you said and let extend the pr #5496 to specify also a `disabled` constructor and have it deprecated in the core.


see commit  https://github.com/neos/neos-development-collection/pull/5463/commits/632b21690b0948e23d27c443219e6d62665716f0

Reply:
The idea of a `SoftRemovedTag` came from the wrapper objects `ContentStreamEventStreamName` and `WorkspaceEventStreamName` where we could have had also a `ContentRepositoryEventStream` object or something...

I guess a `SoftRemovedTag` and a`DeletedTag` objects are not too bad and can actually be nicely encapsulated into their own contexts/ folders but having a  `NeosSubtreeTags` seems indeed the better choice to have it all in one place